### PR TITLE
Bump bitsandbytes minimum version

### PR DESCRIPTION
**Pull request by kashif**
## What does this PR do?

Check to see if the new property can be used else revert to the older API

needed for https://github.com/Lightning-AI/lit-gpt/pull/946

----
📚 Documentation preview 📚: https://pytorch-lightning--19520.org.readthedocs.build/en/19520/

### CODE REVIEW

File: requirements/fabric/strategies.txt [hunk 1]
```diff
@@ -6,4 +6,4 @@
 # note: is a bug around 0.10 with `MPS_Accelerator must implement all abstract methods`
 #  shall be resolved by https://github.com/microsoft/DeepSpeed/issues/4372
 deepspeed >=0.8.2, <=0.9.3; platform_system != "Windows"  # strict
-bitsandbytes ==0.41.0  # strict
+bitsandbytes>0.41.0
```
Comment (Borda) [inline requirements/fabric/strategies.txt:9]: There has to be upper bound, so the actual latest

**Review by Borda — DISMISSED** [review]

**Comment by kashif** [comment]
@Borda are you stalking me 🤪

**Review by Titus-von-Koeller — APPROVED** [review]
Thanks everyone, really appreciated! Everything looks good to me.

File: requirements/fabric/strategies.txt [hunk 2]
```diff
@@ -6,4 +6,4 @@
 # note: is a bug around 0.10 with `MPS_Accelerator must implement all abstract methods`
 #  shall be resolved by https://github.com/microsoft/DeepSpeed/issues/4372
 deepspeed >=0.8.2, <=0.9.3; platform_system != "Windows"  # strict
-bitsandbytes ==0.41.0  # strict
+bitsandbytes>=0.42.0
```
Comment (awaelchli) [inline requirements/fabric/strategies.txt:9]:
```suggestion
bitsandbytes<0.43.0
```

File: requirements/pytorch/extra.txt [hunk 3]
```diff
@@ -8,4 +8,4 @@ hydra-core >=1.0.5, <1.4.0
 jsonargparse[signatures] >=4.27.5, <4.28.0
 rich >=12.3.0, <13.6.0
 tensorboardX >=2.2, <2.7.0  # min version is set by torch.onnx missing attribute
-bitsandbytes ==0.41.0  # strict
+bitsandbytes>=0.42.0
```
Comment (awaelchli) [inline requirements/pytorch/extra.txt:11]:
```suggestion
bitsandbytes<0.43.0
```

**Review by awaelchli — APPROVED** [review]

**Review by Borda — DISMISSED** [review]

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]
Hoping that this is everything that needs an update. Tests seem to agree with htis.

**Pull request merged**